crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -855,4 +867,64 @@ where
             pipe_from_stream(pending, mapping_stream).boxed(),
         );
     }
+
+    fn subscribe_filtered_object_mappings(
+        &self,
+        pending: PendingSubscriptionSink,
+        hashes: Vec<Blake3HashHex>,
+    ) {
+        if hashes.len() > MAX_OBJECT_HASHES_PER_SUBSCRIPTION {
+            warn!(
+                "Request hash count ({}) exceed the server limit: {} ",
+                hashes.len(),
+                MAX_OBJECT_HASHES_PER_SUBSCRIPTION
+            );
+
+            let err_fut = pending.reject(ErrorObject::owned(
+                // TODO: is there a defined error code for an internal request limit?
+                1,
+                format!(
+                    "Request hash count ({}) exceed the server limit: {} ",
+                    hashes.len(),
+                    MAX_OBJECT_HASHES_PER_SUBSCRIPTION
+                ),
+                None::<()>,
+            ));
+
+            self.subscription_executor.spawn(
+                "subspace-filtered-object-mappings-subscription",
+                Some("rpc"),
+                err_fut.boxed(),
+            );
+
+            return;
+        };
+
+        let hashes: HashSet<Blake3Hash> = HashSet::from_iter(hashes.into_iter().map(|hash| *hash));
+
+        // The genesis segment isn't included in this stream, see
+        // `subscribe_archived_object_mappings` for details.
+        let mapping_stream = self
+            .archived_segment_notification_stream
+            .subscribe()
+            .flat_map(move |archived_segment_notification| {
+                let objects: Vec<GlobalObject> = archived_segment_notification
+                    .archived_segment
+                    .global_object_mappings()
+                    .filter(|object| hashes.contains(&object.hash))
+                    .collect();
```

Comment:
In most cases in the codebase we place the type at `.collect()` rather than variable definition:
```suggestion
                let objects = archived_segment_notification
                    .archived_segment
                    .global_object_mappings()
                    .filter(|object| hashes.contains(&object.hash))
                    .collect::<Vec<_>>();
```

Also `stream::iter()` below works with iterators, so you probably don't need to collect it at all. Not only `global_object_mappings` clones data internally, this also does allocation now. Just remove `.collect()` and it should still work.

Reply:
It doesn’t work, because the filter takes ownership of hashes, which belongs to this method. So the compiler says isn’t allowed to escape via stream::iter().

Happy to try another solution, maybe using two moves will help.

Reply:
I think collect is required to avoid this ownership error:
```
error: captured variable cannot escape `FnMut` closure body
   --> crates/sc-consensus-subspace-rpc/src/lib.rs:917:17
...
917 |                 stream::iter(objects)
    |                 ^^^^^^^^^^^^^^^^^^^^^ returns a closure that contains a reference to a captured variable, which then escapes the closure body
    |
    = note: `FnMut` closures only have access to their captured variables while they are executing...
    = note: ...therefore, they cannot allow references to captured variables to escape
```